src/controllers/TalksController.php

```diff
@@ -140,8 +142,229 @@ public function postAction($request, $db) {
+
+            $talk['talk_title'] = filter_var($request->getParameter("talk_title"), FILTER_SANITIZE_STRING);
+            if(empty($talk['talk_title'])) {
+                $errors[] = "'talk_title' is a required field";
+            }
+
+            $talk['url_friendly_talk_title'] = filter_var($request->getParameter("ur_friendly_talk_title"), FILTER_SANITIZE_STRING);
+            if(empty($talk['url_friendly_talk_title'])) {
+                $talk['url_friendly_talk_title'] = $talk['talk_title'];
+            }
+
+            $talk['talk_description']  = filter_var($request->getParameter("talk_description"), FILTER_SANITIZE_STRING);
+            if (empty($talk['talk_description'])) {
+                $errors[] = "'talk_description' is a required field";
+            }
+
+            $talk['type']  = filter_var($request->getParameter("type"), FILTER_SANITIZE_STRING);
+            if (empty($talk['type'])) {
+                $errors[] = "'type' is a required field";
+            }
+            // TODO @heiglandreas check whether the type already exists in the database.
+
+            if ($errors) {
+                throw new Exception(implode(". ", $errors), 400);
+            }
+
+
+            $talk['start_date'] = (new \DateTime($request->getParameter("start_date")))->format('U');
+
+            $talk['duration'] = filter_var(
+                $request->getParameter('duration'),
+                FILTER_SANITIZE_NUMBER_INT
+            );
+            if (empty($talk['duration'])) {
+                $talk['duration'] = 60;
+            }
+
+            $talk['language'] = filter_var($request->getParameter('language'), FILTER_SANITIZE_STRING);
+            if (empty($talk['language'])) {
+                $talk['language'] = 'English - UK';
+            }
+            // TODO: @heiglandreas Check whether the language actually exists in the database
+            // When the language doesn't exist, the talk will not be found
+
+            $talk['slides_link'] = filter_var($request->getParameter('slides_link'), FILTER_SANITIZE_URL);
+
+            $incoming_speakers_list = $request->getParameter('speakers');
+            if(is_array($incoming_speakers_list)) {
+                $talk['speakers'] = array_map(function($speaker){
+                    $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                    $speaker = trim($speaker);
+                    return $speaker;
+                }, $incoming_speakers_list);
+            }
+
+            $talk_mapper = new TalkMapper($db, $request);
+            $new_id = $talk_mapper->save($talk);
+
+            $uri = $request->base . '/' . $request->version . '/talks/' . $new_id;
+            header("Location: " . $uri, true, 201);
+            exit;
+        //    throw new Exception("method not yet supported - sorry");
+        }
+    }
+
+    /**
+     * Edit a talk
+     *
+     * This action expects the following parameters:
+     *
+     * * int talk_id
+     * * int event_id
```

Comment:
`talk_id` and `event_id` are not parameters when editing a talk. This is because we are PUTing to the talk resource directly, so the URI already has enough information in it for us to identify the talk and event ids.